emulator/src/uartlite.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "mmio_dev.h"
+#include <cstring>
+#include <algorithm>
+#include <queue>
+#include <mutex>
+
+#define SR_TX_FIFO_FULL         (1<<3) /* transmit FIFO full */
```

Comment:
Looks like our simple uart implementation is never full on rx or tx, add a comment to note here.